pkg/registration/spoke/addon/registration_controller.go

```diff
@@ -213,6 +215,16 @@ func (c *addOnRegistrationController) startRegistration(ctx context.Context, con
 	additonalSecretData := map[string][]byte{}
 	if config.registration.SignerName == certificatesv1.KubeAPIServerClientSignerName {
 		additonalSecretData[clientcert.KubeconfigFile] = c.kubeconfigData
+	} else {
+		// TODO: this is a workaround for issue-10093. it doesn't contains in the future MCE release(2.6+)
+		// Decode CA from kubeconfigData, hash it and store in the additionalSecretData
+		config, err := clientcmd.Load(c.kubeconfigData)
+		if err != nil {
+			klog.Error(err, "Failed to load kubeconfig, skip adding hubHash to additionalSecretData")
+		} else {
+			hash := sha256.Sum256(config.Clusters["default-cluster"].CertificateAuthorityData)
```

Comment:
The import-controller generates the bootstrap hub kubeconfig with cluster name `default-cluster`, but I'm not sure if there is any case where the bootstrap hub kubeconfig has a different cluster name. In that case, the line will fail.

Reply:
Agree, updated to avoid potential name confictions.